core/src/main/scala/cats/instances/StaticMethods.scala

```diff
@@ -60,4 +60,14 @@ private[cats] object StaticMethods {
     }
   }
 
+  def mapWithLongIndexFromStrictFunctor[F[_], A, B](fa: F[A], f: (A, Long) => B)(implicit ev: Functor[F]): F[B] = {
```

Comment:
Thanks for adding this! The next step is to use it to provide overrides of `mapWithLongIndex` in all the same places that the `Int`-version is used.
https://github.com/typelevel/cats/search?q=mapWithIndexFromStrictFunctor

Reply:
Should be done in https://github.com/typelevel/cats/pull/4247/commits/6fe035f533b7ff1d46eb297c4b2d6b9954dcdd53. I don't know if I should update the tests too, or these implementations get used automatically in some tests.